Some more contributions (#18)
CaptainSifff/paper_teaching-learning-RSE: paper_teaching-learning-RSE.md

and that use of third party software might restrict licensing of the resulting work.

#### Software Behaviour Awareness and Analysis(MOD)
By this, we mean a certain quality of analytical thinking that enables you to form a mental model of the piece of software under consideration in the current environment.

> Using that an RSE should be able to make predictions about a software's behaviour. This is a required skill for tasks like debugging, profiling, designing good tests, or predicting user interaction.

Reviewer: I think a comma after "Using that" is missing, right?

Reviewer 2: Me, too.

a vital role in quality research software.

#### ${DOMAIN}-RSE
While software is the lingua franca of all RSEs there will be RSEs that have specialized in the initricacies of one particular research domain, such as medical RSEs, digital humanities RSEs or physics RSEs.

### Optional RSE competencies -> Maintenance RSEs 

> Oftentimes, a significant amount of effort in (research) software development needs to be spent on maintenance to ensure that software remains useful for researchers now and in the future. The research environment is constantly changing and this can also apply to the software requirements. Accordingly, software often needs to be adapted continuously. If it isn't the software can reach a point where it simply isn't useful to the researchers anymore. To avoid this, regular work needs to be invested. 

Reviewer: Comma after "If it isn't"?

Reviewer: I think [CAA] (https://caa-international.org/about/) might not approve of the suggested term

> While ensuring maintenance and sustainability of research software is of huge importance to the communities that build and use it, a particular challenge is that it's often very difficult to obtain ongoing research funding for software maintenance tasks. As a result, when a project that developed or extended a piece of software finishes, it can often be the case that support for the software fades as team members move on to other research, academic or RSE roles, or become busy with other funded work. While this is not a core concern of this paper, we wanted to highlight this important issue that is frequently faced when working with software in the research communituy.

Reviewer: small typo, communituy should be community

### Organization of teaching

#### Certificates
With the ever-growing demand for RSEs in science, it is helpful for the job-market that people can earn the respective certificates.
Certainly, a big demand for specialized RSEs will come from the newly established RSE centers at research institutions that require skilled people to fill their vacant positions.
Then this demand can find its market with people offering this skill.
Having these certificates provides finally a clear understanding of which tasks an RSE can perform and thereby helps defining the career path and the job description.

#### A possible graduation path within the classical university structures

> We have put forward the idea that familiarity with research is a prerequisite for an RSE in order to fit into the research space.

Reviewer 3: ```suggestion
We have put forward the idea that familiarity with research is a prerequisite for an RSE in order to be able to work effectively in the research space and in collaboration with researchers. In this particular example, we consider a path into RSE via a traditional university route involving Bachelors and Masters degree studies that include an RSE element. However, we recognise that there are other routes into an RSE career and these are increasing. For example, some RSEs come from an industry background, others may come through apprenticeship or similar programmes. In both cases, gaining knowledge of the research lifecycle and understanding the ways that researchers work towards solutions to research challenges is something that can be developed on-the-job alongside training opportunities and the chance to work directly with researchers.
```

> This leads naturally to the question, whether it is possible to become an RSE without a home research domain. 

Reviewer 3: Suggested some changes in the sections below. One point addressed in a couple of places is that people can graduate at any age and, indeed, become RSEs at any age - e.g. they may be retraining from another career path. I therefore think that inexperienced would be a better term so I've addressed this in the sections below.

Reviewer 3: Just a follow up to the comment above...as I've gone through this section making comments, I'm thinking that it feels increasingly contrived. I think it's making an argument that, at a general level, is not entirely valid. I think I understand what this is trying to cover and I've attempted to add some more context into the description.

Nonetheless, I can think of many highly-effective RSEs who don't have PhDs but who work and communicate extremely well with researchers and are ultimately able to produce very high quality research software outputs.

Reviewer: I agree, it's a fine line between "how it currently is" and "how it should/could be". Nevertheless, taking current funding policies into account is a good thing to note here (and was already done). For example you can only hand in proposals to the DFG if you have a PhD...

Reviewer 2: I've added that to the main branch... forgot we had that discussion over here...

> It is of course not possible for everybody to do research into the research field of RSE, therefore we consider it much more useful if young graduates perform their research in a specific domain and thereby have a natural home domain.

Reviewer 3: ```suggestion
With software being a core element of the research process in so many different domains, it is not helpful for everybody working in RSE to have a background in computing or software engineering. Indeed, we consider it much more useful if new graduates looking to work in the RSE space come from a wide range of different domains. Expertise, beyond software development skills, in another research domain can be an important element of an RSE team being able to support RSE projects in that domain.
```

Assuming for a moment, that people have done their masters studies in a particular domain e.g. from the natural sciences and that we can assume that the lectures to that point contain a mixture of domain specific content and RSE specific content (A good starting point for an RSE in the natural sciences), then we come to the question of the topic of the masters thesis.

> In order for young RSEs to get their research experience we believe it is necessary that already in their master's thesis young RSE students are given computational research tasks that can be solved with the RSE specific skills of that domain. 

Reviewer 3: ```suggestion
In order for new and inexperienced RSEs to gain research experience, we believe it is necessary that students are given computational research tasks, as part of masters thesis work, that can be solved with the RSE specific skills of that domain. 
```

> This gives them a Master's degree of a ${DOMAIN}-RSE that has learnt in their lectures a research domain specific part and a software engineering specific part, and enabled them to get a first dip into actual science in their master's thesis.

Reviewer 3: ```suggestion
This gives them a Master's degree of a ${DOMAIN}-RSE where the studies have included lectures covering research domain-specific topics alongside material covering software engineering skills and research software best practices. This enabled the student to get initial experience of real research work during their Master's thesis.
```

> Of course, the next question for their future is whether a master's degree enables them to really be effective parts of a research group.

Reviewer 3: ```suggestion
Of course, the next question for their future is whether a Master's degree serves as the only prerequisite needed to become an effective member of a research group.
```

> We argue that this is most likely not the case since they will not be accepted into research groups and there are formal barriers in the research landscape if people don't have a PhD (e.g. eligibility for funding).

Reviewer 3: ```suggestion
While we accept this is something of a generalisation, we argue that this is most likely not the case since undertaking a PhD provides a much more extensive set of research training and experience that can be vital for a researcher's future. Research environments different internationally but in many cases there are formal barriers in the research landscape that require a PhD (e.g. eligibility for funding).
```

> Hence a PhD is required to actively participate in science and why we argue the regular RSE should do a PhD that on one end combines knowledge of a research domain with software engineering heavy task such that both pillars are suitably covered and they were able to observe how research functions.

Reviewer 3: Hence, in the context of this specific path, we consider that undertaking a PhD is an important element in being able to develop in the research environment and it's something that we argue research-focused RSEs should do. More specifically, we recommend undertaking a PhD project that combines domain research with large-scale computation in order to develop a strong research base around both the home domain and RSE.

Reviewer: I'm not sure whether "large-scale computation" fits in every domain. To me it sounds a bit too much like HPC which is a very STEM thing and even there mostly EM + Physics.

#### Specialised Master's Programs

> On the other hand, when pursuing a PhD, scientists are more and more required to do RSE work as part of their research. 

Reviewer 3: ```suggestion
On the other hand, when pursuing a PhD, scientists are increasingly required to do RSE-type work as part of their research as data and computation are becoming part of research tasks in a huge range of fields.
```

> It is not uncommon for scientists to be faced with RSE topics for the first time, because it has not been part of their academic curricula.

Reviewer 3: ```suggestion
It is not uncommon for researchers to be faced with RSE topics for the first time, because it has not been part of their academic curricula.
```

Author: Noticed that you (@jcohen02) prefer the term researcher over scientists. Need to ask you tomorrow about the difference...

Reviewer 3: 🙂 Happy to discuss tomorrow but basically, I feel that software is relevant in so many domains and if we consider areas such as engineering, business, arts and humanities, for example, I don't think scientist is the right term to describe researchers in these fields? e.g. I think a Civil Engineering researcher would class themselves as an engineer but not a scientist? I see "researcher" as the most generic term for anyone undertaking research in any domain. Happy to change the terminology being used here if you prefer.

Many are faced with a steep learning curve that requires them to invest a huge amount of time to catch up.
Naturally, many would only invest as much as necessary to get the job done regardless of whether the solution is sustainable or not.

> Support from RSEs is one way to find a way out. 

Reviewer 3: ```suggestion
Support from RSEs is one way to resolve this challenge. 
```

> Another would be to lay the foundation for future RSE work already early in their curricula.

Reviewer 3: ```suggestion
Another would be to lay more effective foundations for future RSE work at a much earlier stage in undergraduate/postgraduate studies.
```

> Ideally, even establish dedicated master programs which specialise in RSE work.

Reviewer 3: ```suggestion
We see scope for establishing dedicated RSE Master's programmes which specialise in developing RSE skills and practices.
```

Some universities already offer dedicated master's programs in some domains. 

> Examples would be Computational Sciences in Engineering (CSE) or Bio-Informatics. 

Reviewer 3: ```suggestion
Examples would be Computational Sciences in Engineering (CSE) or Bioinformatics. 
```

Assuming for a moment, that people have done their masters studies in a particular domain e.g. from the natural sciences and that we can assume that the lectures to that point contain a mixture of domain specific content and RSE specific content (A good starting point for an RSE in the natural sciences), then we come to the question of the topic of the masters thesis.
In order for young RSEs to get their research experience we believe it is necessary that already in their master's thesis young RSE students are given computational research tasks that can be solved with the RSE specific skills of that domain. 
This gives them a Master's degree of a ${DOMAIN}-RSE that has learnt in their lectures a research domain specific part and a software engineering specific part, and enabled them to get a first dip into actual science in their master's thesis.
Of course, the next question for their future is whether a master's degree enables them to really be effective parts of a research group.

> While we accept this is something of a generalisation, we argue that this is most likely not the case since undertaking a PhD provides a much more extensive set of research training and experience that can be vital for a researcher's future. Research environments different internationally but in many cases there are formal barriers in the research landscape that require a PhD (e.g. eligibility for funding).

Reviewer: Should it not be "Research environments differ internationally, "?
